app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/MongoSendTask.java

```diff
@@ -16,19 +17,42 @@
  */
 public class MongoSendTask extends AsyncTask<String, Void, Void> {
         private Context context;
-        public List<BgSendQueue> bgsQueue = new ArrayList<BgSendQueue>();
+        //public List<BgSendQueue> bgsQueue = new ArrayList<BgSendQueue>();
         public List<CalibrationSendQueue> calibrationsQueue = new ArrayList<CalibrationSendQueue>();
 
+        PowerManager.WakeLock wakeLock;
+        private static int lockCounter = 0;
         private Exception exception;
         private static final String TAG = MongoSendTask.class.getSimpleName();
 
         public MongoSendTask(Context pContext) {
-            calibrationsQueue = CalibrationSendQueue.mongoQueue();
-            bgsQueue = BgSendQueue.mongoQueue();
             context = pContext;
+            PowerManager pm = (PowerManager) context.getSystemService(Context.POWER_SERVICE);
+            wakeLock = pm.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK, "WifiReader");
+            wakeLock.acquire();
+            lockCounter++;
+            Log.e(TAG,"MongosendTask - wakelock acquired " + lockCounter);
+
         }
 
         public Void doInBackground(String... urls) {
+            boolean sendMore = true;
+            try {
+                while (sendMore) {
```

Comment:
This might lead to aggressively uploading data and block the whole internet connection and may lead to additional costs if someone goes online out of the house just to check a website.
The service that starts this task will be called every few minutes anyway so there is no need to continue.

A function that can upload month of data at once should have to be triggered by the user (Button in the menu or settings e.g.)

At leas we should check that this does not happen on mobile data but just on Wifi.